### PR TITLE
US26844: Python SDK - sorted input tool sample plugin

### PR DESCRIPTION
**Pull request by ghost**
initial commit of US26844, also minor formatting changes to US26826

### CODE REVIEW

File: PySortedInputToolExample/PySortedInputToolExampleEngine.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+        A non-interface method.
+        Responsible for building out the proper XML string format for pre_sort.
+        :param element: SortInfo or FieldFilterList
+        :param subelement: The user selected field
+        :param order: Asc or Desc
+        :return: Void
+        """
+        # Building the XML string to pass as an argument to pre_sort's sort info parameter.
+        root = Et.Element(element)
+        sub_element = 'Field field="{0}" order="{1}"' if order != "" else 'Field field="{0}"'
```
Comment (alteryxMChadwick) [inline PySortedInputToolExample/PySortedInputToolExampleEngine.py:41]: Does this format of `if` assignment look alright to the rest of the team?

File: PySortedInputToolExample/PySortedInputToolExampleEngine.py [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+        Handles type validation, and renames the outgoing field.
+        Called when the incoming connection's record metadata is available or has changed, and
+        has let the Alteryx engine know what its output will look like.
+        :param record_info_in: A RecordInfo object containing the XML representation for the incoming connection's field and sort properties.
+        :return: False if type validation fails.
+        """
+        #  Evaluate field type of selected field to make sure only numeric field types are selected
+        if 'String' in str(self.get_single_field_type(record_info_in))\
+                or 'Bool' in str(self.get_single_field_type(record_info_in)):
+            self.output_message('pi_push_all_records', Sdk.EngineMessageType.error, 'Selected A Non-Numeric Field')
```
Comment (alteryxMChadwick) [inline PySortedInputToolExample/PySortedInputToolExampleEngine.py]: There are more field types than just String, Bool, and Numeric...

File: PySortedInputToolExample/PySortedInputToolExampleEngine.py [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+        """
+        if not self.initialized:
+            return False
+        self.record_cnt += 1
+        if self.record_cnt == 1:
+            self.record_creator.reset()
+            self.record_copier.copy(self.record_creator, in_record)
+            out_record = self.record_creator.finalize_record()
+            # Push the record and quit if there's a downstream error
+            if not self.output_anchor.push_record(out_record):
```
Comment (alteryxMChadwick) [inline PySortedInputToolExample/PySortedInputToolExampleEngine.py]: `return self.output_anchor.push_record(out_record)`

Comment (alteryxMChadwick) [reply PySortedInputToolExample/PySortedInputToolExampleEngine.py]: Actually, if you're only pushing the first record down, try just always `return False`

File: PySortedInputToolExample/PySortedInputToolExampleEngine.py (hunk 2 above)
Comment (ghost) [reply PySortedInputToolExample/PySortedInputToolExampleEngine.py]: it's what I get for rushing! hopefully that fix does it

**Pull request merged**